osu.Game/Skinning/Skin.cs

```diff
@@ -230,6 +205,81 @@ public void UpdateDrawableTarget(SkinComponentsContainer targetContainer)
             return null;
         }
 
+        #region Deserialisation & Migration
+
+        private SkinLayoutInfo? parseLayoutInfo(string jsonContent, SkinComponentsContainerLookup.TargetArea target)
+        {
+            SkinLayoutInfo? layout = null;
+
+            // handle namespace changes...
+            jsonContent = jsonContent.Replace(@"osu.Game.Screens.Play.SongProgress", @"osu.Game.Screens.Play.HUD.DefaultSongProgress");
+            jsonContent = jsonContent.Replace(@"osu.Game.Screens.Play.HUD.LegacyComboCounter", @"osu.Game.Skinning.LegacyComboCounter");
+            jsonContent = jsonContent.Replace(@"osu.Game.Screens.Play.HUD.PerformancePointsCounter", @"osu.Game.Skinning.Triangles.TrianglesPerformancePointsCounter");
```

Comment:
I digress but I wonder how long _this_ crap will have to live... I'd hope we can remove it in a few months or something.

Reply:
Well, technically this can live on as part of the migration process for 100% compatibility, I was just unable to move it into the actual migration method because it manipulates the JSON string itself before deserialising it.